### ISSUE TITLE
Startup profiling

### ISSUE THREAD

**Issue by rozyczko** [easyScience/easyDiffractionApp#193]

**Comment by rozyczko** [comment]
Analysis with py-spy shows three main culprits

easyCore
==========
```
easyCore\easyCore\__init__.py:11
import pint
```
takes 6 sec. This import can likely be folded into one of the object initializers


```
easyCore\easyCore\Fitting\__init__.py:11
from lmfit import Parameter as lmParameter, Parameters as lmParameters, Model as lmModel
```
takes ~4 sec.
This can also be moved to the following `__init__`

cryspy
======
```
cryspy\cryspy\A_functions_base\function_3_mcif.py:718
read_magnetic_data()
```
takes ~ 4sec

needs closer inspection

**Comment by rozyczko** [comment]
This startup analysis file should be opened by the speedscope app:
https://www.speedscope.app/

[startup_timings.log](https://github.com/easyScience/easyDiffractionApp/files/8041348/startup_timings.log)

**Comment by rozyczko** [comment]
On average, importing cryspy takes 4.25s while GSASII takes about 1.5s (locally).

It would be good to replace the calculator check in `easyDiffractionLib\easyDiffractionLib\Interfaces\__init__.py` with a check that does not directly import those modules. However, this check also does quite a bit of setting up the interfaces, so it won't be as easy. Postponing the actual calculator import, e.g. in `Calculators\cryspy.py:7` would also be an option.

**Comment by rozyczko** [comment]
Loading `cryspy` can't be made lazy since the calculators are all **instantiated** on startup.

**Comment by rozyczko** [comment]
Also, improved performance by making the profile update run only once for
- project load
- experiment cif load
- parameter change

This should be helpful
